packages/react-chat/src/components/Modals/StatusModal.tsx

```diff
@@ -0,0 +1,171 @@
+import QRCode from "qrcode.react";
+import React, { useState } from "react";
+import styled from "styled-components";
+
+import { buttonStyles } from "../Buttons/buttonStyle";
+import { PasteInput } from "../Form/PasteInput";
+import { MobileIcon } from "../Icons/MobileIcon";
+import { ProfileSvg } from "../Icons/ProfileIcon";
+import { ScanIcon } from "../Icons/ScanIcon";
+import { textMediumStyles } from "../Text";
+
+import { Modal } from "./Modal";
+import { Heading, Section } from "./ModalStyle";
+
+export const StatusModalName = "StatusModal";
+
+export enum StatusModalState {
+  Mobile,
+  Desktop,
+}
+
+export function StatusModal() {
```

Comment:
This looks like a really big component, maybe let's try to split this into few smaller ones, even in the same file.